richtextfx/src/main/java/org/fxmisc/richtext/ParagraphBox.java

```diff
@@ -125,7 +125,9 @@ public String toString() {
                         return Optional.empty();
                     }
                 });
-        EventStream<?> stops = stationaryEvents.filter(Either::isRight).map(Either::getRight);
+        EventStream<?> stops = stationaryEvents.filter(Either::isRight).map(Either::getRight)
+                // only fire END event when a BEGIN event has already been fired
+                .conditionOn(beginEventFired);
```

Comment:
This line should occur immediately after stationaryEvents to speed up the process